openslides/core/views.py

```diff
@@ -338,3 +344,20 @@ def check_permissions(self, request):
         if (self.action in ('create', 'update', 'destroy') and
                 not request.user.has_perm('core.can_manage_tags')):
             self.permission_denied(request)
+
+
+class UrlPatternsView(utils_views.APIView):
+    """
+    Returns a dictonary with all url patterns as json.
+    """
+    URL_KWARGS_REGEX = re.compile(r'%\((\w*)\)s')
+    http_method_names = ['get']
+
+    def get_context_data(self, **context):
+        result = {}
+        url_dict = get_resolver(None).reverse_dict
+        for pattern_name in filter(lambda key: isinstance(key, str), url_dict.keys()):
+            url = url_dict[pattern_name][0][0][0]
```

Comment:
This line is very complex and would be difficult to debug. Can you please split it up to make more clear was these multiple access on the nested list does?